pkgs/applications/networking/cloudflared/default.nix

```
@@ -17,22 +17,42 @@ buildGoModule rec {
 
   preCheck = ''
     # Workaround for: sshgen_test.go:74: mkdir /homeless-shelter/.cloudflared: no such file or directory
-    export HOME="$(mktemp -d)";
+    export HOME="$(mktemp -d)"
 
     # Workaround for: protocol_test.go:11:
     #   lookup protocol-v2.argotunnel.com on [::1]:53: read udp [::1]:51876->[::1]:53: read: connection refused
-
     substituteInPlace "edgediscovery/protocol_test.go" \
       --replace "TestProtocolPercentage" "SkipProtocolPercentage"
+
+    # Workaround for: origin_icmp_proxy_test.go:46:
+    #   cannot create ICMPv4 proxy: socket: permission denied nor ICMPv6 proxy: socket: permission denied
+    substituteInPlace "ingress/origin_icmp_proxy_test.go" \
+      --replace "TestICMPRouterEcho" "SkipICMPRouterEcho"
+
+    # Workaround for: origin_icmp_proxy_test.go:110:
+    #   cannot create ICMPv4 proxy: socket: permission denied nor ICMPv6 proxy: socket: permission denied
+    substituteInPlace "ingress/origin_icmp_proxy_test.go" \
+      --replace "TestConcurrentRequestsToSameDst" "SkipConcurrentRequestsToSameDst"
+
+    # Workaround for: origin_icmp_proxy_test.go:242:
+    #   cannot create ICMPv4 proxy: socket: permission denied nor ICMPv6 proxy: socket: permission denied
+    substituteInPlace "ingress/origin_icmp_proxy_test.go" \
+      --replace "TestICMPRouterRejectNotEcho" "SkipICMPRouterRejectNotEcho"
+
+    # Workaround for: icmp_posix_test.go:28: socket: permission denied
+    substituteInPlace "ingress/icmp_posix_test.go" \
+      --replace "TestFunnelIdleTimeout" "SkipFunnelIdleTimeout"
   '';
 
   doCheck = !stdenv.isDarwin;
 
+  passthru.tests = callPackage ./tests.nix { inherit version; };
```

Everything looks good other than this; I am not sure if this is the right pattern, hopefully someone else can shed some light on that.

---

First time looking into this too. It seems writing tests like this is a [documented pattern](https://ryantm.github.io/nixpkgs/contributing/coding-conventions/#ssec-inline-package-tests-writing) however you should add your own key for the test itself e.g. `passthru.tests.simple`. [home-assistant](https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/home-assistant/default.nix#L361) seems to be a decent example of this.

```suggestion
  passthru.tests.simple = callPackage ./tests.nix { inherit version; };
```

---

makes sense, added your suggestion